lib/galaxy/tools/parameters/basic.py

```diff
@@ -2300,27 +2303,38 @@ def append(list, hda, name, src, keep=False, subcollection_type=None):
             return list.append(value)
 
         def append_dce(dce):
-            if dce.hda:
-                # well this isn't good, but what's the alternative ?
-                # we should be precise about what we're (re-)running here.
-                key = "hda"
-            else:
-                key = "hdca"
-            d["options"][key].append(
+            d["options"]["dce"].append(
                 {
                     "id": trans.security.encode_id(dce.id),
                     "name": dce.element_identifier,
+                    "hda": dce.hda,
```

Comment:
Unless you're using this to determine if this represents a leaf in a collection this doesn't look right. If you do then this should just be a boolean.

Reply:
Thanks for looking into this, and yes I am using this to determine if it's a dataset, and I think it's ok to keep it as `dce.hda`, feels more consistent at first sight.

Reply:
I switched it over to a boolean named `is_dataset` as suggested.